### ISSUE TITLE
Trie::get() requires ownership of key values

### ISSUE THREAD

**Issue by Kylebrown9** [Randevelopment/Slang#4]
Trie::get() has this behavior because TrieView::descend() also requires ownership.

Approach A: make TrieView::descend(), not require ownership and make HashTrieView never dereference.
Approach B: use Clone or ToOwned to take non-owned key values and make owned ones

**Comment by Kylebrown9** [comment]
Another way to think of it is that TrieView::descend() currently requires ownership because current implementers need to create a tuple from the key value and an index. 

A HashMap-like type which takes two key types and allows get using references to the two values in separate locations (as opposed to adjacently in a tuple) would keep HashTrieView(Mut) from needing to dereference the key, to colocate it.

**Comment by Kylebrown9** [comment]
Note to self:
Attempt implementation based on [this](https://stackoverflow.com/questions/45786717/how-to-implement-hashmap-with-two-keys)

**Comment by Kylebrown9** [comment]
In working on this, I ran into issues communicating, the way that view lifetimes interacted, with the borrow checker. I also realized that one option we can take right now, that solves all of this, is to simply implement the intended lifetime behaviors in concrete instead of with traits. This simplifies many things, and when it does add lifetimes and complexities, they directly pay off. 

The resulting implementation does not require ownership for get, can directly return values and does not require ownership when borrowing semantics should be sufficient.
